specs/agents/tracing-spans-destination.md

```diff
@@ -62,23 +75,70 @@ The cardinality should be the same or higher as `span.destination.service.name`.
 Higher, if there are individual sub-resources for a service, such as individual queues for a message broker.
 Same cardinality otherwise.
 
+**API**
+
+Agents SHOULD offer a public API to set this field so that users can customize the value if the generic mapping is not sufficient.
+User-supplied values MUST have the highest precedence.
+
+To allow for automatic inference,
+without users having to specify any destination field,
+agents SHOULD offer a dedicated API to start an exit span.
+This API sets the `exit` flag to `true` and returns `null` or a noop span in case the parent already represents an `exit` span.
+
 **Value**
 
-Usually, the value is just the `span.subtype`.
-For HTTP, this is the host and port (see the [HTTP spec](tracing-instrumentation-http.md#destination) for more details).
-The specs for the specific technologies will have more information on how to construct the value for `context.destination.service.resource`.
+For all exit spans,
+agents MUST infer the value of this field based on properties that are set on the span.
+
+This is how to determine whether a span is an exit span:
+```groovy
+exit = exit || context.destination || context.db || context.message
+```
+
+For each exit span that does not have a value for `context.destination.service.resource`,
+agents MUST run this logic to infer the value.
+```groovy
+if      (context.http?.url)            "${context.http.url.host}:${context.http.url.port}"
```

Comment:
Should `context.http.url` come last (before the final catch-all "else"), for cases where the service is a database or messaging service which provides an HTTP-based API? i.e. for spans which have both db & http context, I would expect db to take precedence.

Reply:
Good catch